components/encryption/src/io.rs

```diff
@@ -17,70 +17,162 @@ use file_system::File;
 use tikv_util::box_err;
 
 const AES_BLOCK_SIZE: usize = 16;
-const MAX_CRYPTER_READER_BUFFER_SIZE: usize = 1024;
+const MAX_INPLACE_CRYPTION_SIZE: usize = 1024 * 1024;
 
-/// Encrypt content as data being read.
-pub struct EncrypterReader<R>(CrypterReader<R>);
+/// Encrypt or decrypt content as data being read.
+pub struct CrypterReader<R> {
```

Comment:
can we still separate `Encrypter` and `Decrypter`?

Reply:
Make sense, it's separated now.